From 64150cc7c5ed60bf6bc422908a1cc39fccde2fe0 Mon Sep 17 00:00:00 2001
From: Holger Weiss <holger@zedat.fu-berlin.de>
Date: Sun, 13 Aug 2017 20:31:03 +0200
Subject: [PATCH] Let 'domain_certfile' take higher precedence

If a 'domain_certfile' is specified, use that instead of the
's2s_certfile' (or 'c2s_certfile').
---
 src/ejabberd_c2s.erl | 4 ++--
 src/ejabberd_s2s.erl | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

Index: ejabberd/src/ejabberd_c2s.erl
===================================================================
--- ejabberd.orig/src/ejabberd_c2s.erl
+++ ejabberd/src/ejabberd_c2s.erl
@@ -303,9 +303,9 @@ tls_options(#{lserver := LServer, tls_op
 		   {true, CertFile} when CertFile /= undefined -> DefaultOpts;
 		   {_, _} ->
 		       case ejabberd_config:get_option(
-			      {c2s_certfile, LServer},
+			      {domain_certfile, LServer},
 			      ejabberd_config:get_option(
-				{domain_certfile, LServer})) of
+				{c2s_certfile, LServer})) of
 			   undefined -> DefaultOpts;
 			   CertFile -> lists:keystore(certfile, 1, DefaultOpts,
 						      {certfile, CertFile})
Index: ejabberd/src/ejabberd_s2s.erl
===================================================================
--- ejabberd.orig/src/ejabberd_s2s.erl
+++ ejabberd/src/ejabberd_s2s.erl
@@ -199,9 +199,9 @@ dirty_get_connections() ->
 -spec tls_options(binary(), [proplists:property()]) -> [proplists:property()].
 tls_options(LServer, DefaultOpts) ->
     TLSOpts1 = case ejabberd_config:get_option(
-		      {s2s_certfile, LServer},
+		      {domain_certfile, LServer},
 		      ejabberd_config:get_option(
-			{domain_certfile, LServer})) of
+			{s2s_certfile, LServer})) of
 		   undefined -> DefaultOpts;
 		   CertFile -> lists:keystore(certfile, 1, DefaultOpts,
 					      {certfile, CertFile})
